Sources/Breadcrumb.swift

```diff
@@ -8,81 +8,63 @@
-// due to compatibility with Objective-C. These should really be
-// enums with associative values (because much cleaner) but
-// Objective-C does not recognize them.
-
-/// A class used to represent the breadcrumbs attached to events
+/// A class used to represent the breadcrumbs leading up to an events
 @objc public class Breadcrumb: NSObject {
-	
-	public var timestamp: NSDate
+
+	// MARK: - Attributes
+
+	public let timestamp: NSDate
 	public var category: String
-	
+
 	public var type: String?
 	public var message: String?
 	public var data: [String: AnyObject]
-	public var level: SentrySeverity?
-	
-	/// Creates a user
-	public init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, type: String? = nil, level: SentrySeverity? = nil, data: [String: AnyObject]? = nil) {
+	public var level: SentrySeverity // can't be optional because @objc can't handle optional enums
+
+
+	/// Creates a breadcrumb
+	@objc public init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, type: String? = nil, level: SentrySeverity = .Info, data: [String: AnyObject]? = nil) {
 		self.category = category
 		self.timestamp = timestamp
 		self.message = message
 		self.type = type
 		self.data = data ?? [:]
-	}
+		self.level = level
 
-}
-
-extension Dictionary {
-	
-	// Sets the key and value but only if value is non-nil
-	func set(key: Key, value: Value?) -> Dictionary<Key, Value> {
-		guard let value = value else { return self }
-		
-		var newDict = self
-		newDict[key] = value
-		
-		return newDict
+		super.init()
 	}
-	
-}
 
-extension Breadcrumb {
-	
-	public convenience init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, level: SentrySeverity? = nil, data: [String: AnyObject]? = nil, to: String, from: String? = nil) {
-		
-		let navigationData = (data ?? [:])
-			.set("to", value: to)
-			.set("from", value: from)
-		
+	/// Conveneince init for a "navigation" type breadcrumb
+	public convenience init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, level: SentrySeverity = .Info, data: [String: AnyObject]? = nil, to: String, from: String? = nil) {
+		let navigationData: [String: AnyObject] = (data ?? [:])
+		.set("to", value: to)
+		.set("from", value: from)
+
 		self.init(category: category, timestamp: timestamp, message: message, type: "navigation", level: level, data: navigationData)
 	}
-	
-	public convenience init(category: String, timestamp: NSDate = NSDate(), message: String? = nil, level: SentrySeverity? = nil, data: [String: AnyObject]? = nil, url: String, method: String, statusCode: Int? = nil, reason: String? = nil) {
-		
-		let httpData = (data ?? [:])
-			.set("url", value: url)
-			.set("method", value: method)
-			.set("status_code", value: statusCode)
-			.set("reason", value: "reason")
-		
+
+	/// Conveneince init for an "http" type breadcrumb (-999 workaround because @objc can't handle optional Int)
```

Comment:
@dcvz Can it handle an optional `NSNumber`? Not sure better or worse if it can 😬